src/Simulation/Simulators/OpenSystemsSimulator/DataModel/Generator.cs

```diff
@@ -0,0 +1,146 @@
+                        }
+
+                        var propertyName = reader.GetString();
+                        reader.Read();
+
+                        switch (propertyName)
+                        {
+                            case "vectors":
+                                eigenvectors = arrayConverter.Read(
+                                    ref reader, typeof(NDArray), options
+                                );
+                                break;
+
+                            case "values":
+                                eigenvalues = arrayConverter.Read(
+                                    ref reader, typeof(NDArray), options
+                                );
+                                break;
+
+                            default:
+                                throw reader.CurrentState.Exception($"Unknown property {propertyName}.");
+                        }
+
+                    }
+
+                    if (eigenvalues == null)
+                    {
+                        throw reader.CurrentState.Exception($"Generator kind was ExplicitEigenvalueDecomposition, but no eigenvalues were provided.");
+                    }
+                    Debug.Assert(eigenvalues is not null);
+
+                    if (eigenvectors == null)
+                    {
+                        throw reader.CurrentState.Exception($"Generator kind was ExplicitEigenvalueDecomposition, but no eigenvectors were provided.");
+                    }
+                    Debug.Assert(eigenvectors is not null);
+
+                    return (int nQubits) => new ExplicitEigenvalueDecomposition(nQubits, eigenvalues, eigenvectors);
+
+                default:
+                    throw reader.CurrentState.Exception($"Unknown generator kind {kind}.");
+            }
+        });
+    }
+
+    public override void Write(Utf8JsonWriter writer, Generator value, JsonSerializerOptions options)
+    {
+        var arrayConverter = new ComplexArrayConverter();
+
+        using (writer.WriteObject())
+        {
+            writer.WriteNumber("n_qubits", value.NQubits);
+
+            writer.WritePropertyName("data");
+            if (value is ExplicitEigenvalueDecomposition eig)
+            {
+                using (writer.WriteObject())
+                {
+                    writer.WritePropertyName("ExplicitEigenvalueDecomposition");
+                    using (writer.WriteObject())
+                    {
+                        writer.WritePropertyName("values");
+                        arrayConverter.Write(writer, eig.Eigenvalues, options);
+                        writer.WritePropertyName("vectors");
+                        arrayConverter.Write(writer, eig.Eigenvectors, options);
+                    }
+                }
+            }
+            else if (value is UnsupportedGenerator _)
+            {
+                writer.WriteStringValue("Unsupported");
+            }
+        }
+    }
+}
```

Comment:
(May be paranoid, feel free to ignore)
If the param named `value` gains a new value (that is not `ExplicitEigenvalueDecomposition` and is not `UnsupportedGenerator`) then both the `if` (L126) and `else if` (L140) will _fail_ and the line 125 (`writer.WritePropertyName("data")`) will (silently) NOT be followed with the sensible data. Any subsequent output (`writer.Write...`) may result in wrong/unexpected data in the `writer`.
Consider placing something like `else { .. Compile-time failure .. }` after L143.